### ISSUE TITLE
ttnn::atan2 op reverses args compared to torch

### ISSUE THREAD

**Issue by kmitrovicTT** [tenstorrent/tt-metal#14470]
## Summary

`ttnn::atan2` op reverses args compared to torch/numpy/stablehlo.

### Numpy

https://numpy.org/devdocs/reference/generated/numpy.arctan2.html

```
The quadrant (i.e., branch) is chosen so that arctan2(x1, x2) is the signed angle in radians 
between the ray ending at the origin and passing through the point (1,0), and the ray 
ending at the origin and passing through the point (x2, x1). 
(Note the role reversal: the "y-coordinate" is the first function parameter, 
the "x-coordinate" is the second.) 
By IEEE convention, this function is defined for x2 = +/-0 and for either or 
both of x1 and x2 = +/-inf (see Notes for specific values).
```

Example:
```
>>> import numpy as np
>>> a = 1
>>> b = 0
>>> np.arctan2(a, b)
1.5707963267948966
>>> np.arctan2(b, a)
0.0
```

### StableHLO

https://openxla.org/stablehlo/spec#atan2

Example:

```
// %lhs: [0.0, 1.0, -1.0]
// %rhs: [0.0, 0.0, 0.0]
%result = "stablehlo.atan2"(%lhs, %rhs) : (tensor<3xf64>, tensor<3xf64>) -> tensor<3xf64>
// %result: [0.0, 1.57079637, -1.57079637] // [0.0, pi/2, -pi/2]
```

Works same way as numpy.

### Torch

https://pytorch.org/docs/stable/generated/torch.atan2.html

Example:

```
>>> a = torch.Tensor([a])
>>> b = torch.Tensor([b])
>>> torch.atan2(a, b)
tensor([1.5708])
>>> torch.atan2(b, a)
tensor([0.])
```

Works the same way as numpy and stablehlo.

## Proposed fix

In `third_party/tt-metal/src/tt-metal/ttnn/cpp/ttnn/operations/eltwise/binary/device/binary_composite_op.cpp::_atan2` roles of `input_a` and `input_b` should be reversed in 

```cpp
    Tensor result(input_a);
    {
        Tensor atan_input = ttnn::multiply(
            ttnn::abs(input_b, output_mem_config),
            ttnn::reciprocal(ttnn::abs(input_a, output_mem_config), output_mem_config),
            std::nullopt,
            output_mem_config);
        result = ttnn::atan(atan_input, output_mem_config);
    }
```

**Comment by prajaramanTT** [comment]
@kmitrovicTT is this still an open issue ? If not, can you please close this ? Thanks.

**Comment by kmitrovicTT** [comment]
@prajaramanTT As far as I can see, no, nothing was changed.

Hey @umadevimcw, I see that your [angle issue fix](https://github.com/tenstorrent/tt-metal/pull/15213) was making some modifications to this code. Do you maybe know why `ttnn.atan2` op reverses the roles of `input_a` and `input_b`? Please read the issue description if not familiar with what I am talking about.

**Comment by cmaryanTT** [comment]
@umadevimcw - this should be an easy change

**Comment by umadevimcw** [comment]
@cmaryanTT will check and update here

**Comment by umadevimcw** [comment]
@kmitrovicTT Are you asking why the args are reversed ?  or in proposed fix you have added `abs` op do you want me to include that change

**Comment by kmitrovicTT** [comment]
Take note that my proposed fix is old, back from October, code might no longer look like it did, but the reversed args problem is the same. 

I noticed that args were reversed compared to StableHLO op, which is an important common ground in our compiler. So I checked with numpy and torch, as they are hugely popular, and it was reversed compared to them all.

**Comment by umadevimcw** [comment]
@kmitrovicTT Yeah fixed that in this branch and submitted the draft PR #18279. Will submit for review once CI passed
